src/services/services.ts

```diff
@@ -1294,11 +1296,15 @@ namespace ts {
             // the course of whatever called `synchronizeHostData`
             sourcemappedFileCache = createSourceFileLikeCache(host);
 
-            // Make sure all the nodes in the program are both bound, and have their parent
-            // pointers set property.
-            program.getTypeChecker();

```

Comment:
Instead of this why dont you ensure that createSourceFile makes sure that sourceFile created will have their parent pointers set. I think this is what @mhegazy has suggested earlier

Reply:
https://github.com/Microsoft/TypeScript/blob/master/src/services/documentRegistry.ts#L177 this is whre you want to pass true to setParents

Reply:
That was https://github.com/Microsoft/TypeScript/pull/23033/commits/44431c861f986ef63a5b4a436b81fde0a8739b16.  @mhegazy asked me to revert it (https://github.com/Microsoft/TypeScript/pull/23033/commits/99d3054e0e2510c9dad6ec5ec96b6364f19c1e1c).

Reply:
@sheetalkamat that was my original suggestion, but seems like we make assumptions about symbols being bound more often than i thought.
Most of the time this is true anyways.. we should only be calling this function from semantic features.